sci-mathematics/glucas/patches/glucas-2.9.0.patchset

```diff
@@ -0,0 +1,21 @@
+From d618629166a8f96488b22685e99be986af57ab35 Mon Sep 17 00:00:00 2001
+From: nishanth1232 <thumma.nishanth79@gmail.com>
+Date: Fri, 16 Nov 2018 18:53:27 +0000
+Subject: gsetup.c for glucas
+
+
+diff --git a/gsetup.c b/gsetup.c
+index 7f32832..9a3d46c 100644
+--- a/gsetup.c
++++ b/gsetup.c
+@@ -39,7 +39,7 @@ void setup (int nonice)
+ #ifdef __sgi__
+     (void)schedctl(NDPRI, 0, NDPLOMIN);
+ #endif
+-#if !defined(macintosh) && !defined(pccompiler)
++#if !defined(macintosh) && !defined(pccompiler) !defined(_HAIKU_)
```

Comment:
It's line 42 IIRC you need to tackle :) This line doesn't do the trick

Reply:
Strike that, you got the right one :+1:

Reply:
So can I remove !defined(_HAIKU_)?

Reply:
For some reason this isn't working for me on x86_64 the next (using #ifndef) works for me on x86_gcc2 and x86_64
```
#ifndef __HAIKU__
#if !defined(macintosh) && !defined(pccompiler)
    (void)nice(40);
#endif
```

Reply:
@Begasus 

When I edit the patchfile with that, I get the following error:
```
Applying patchset "/boot/home/haikuports/sci-mathematics/glucas/patches/glucas-2.9.0.patchset" ...
error: corrupt patch at line 9
error: could not build fake ancestor
hint: Use 'git am --show-current-patch' to see the failed patch
Command '['git', 'am', '--ignore-whitespace', '-3', '--keep-cr', u'/boot/home/haikuports/sci-mathematics/glucas/patches/glucas-2.9.0.patchset']' returned non-zero exit status 128
```

Reply:
In https://github.com/haikuports/haikuports/pull/3372#issuecomment-439379957 I mentioned that patchset files should not be written but generated.
This also means that if your patchset modifies a file, e.g. gsetup.c, but you would like to change something in the patchset file, what you need to do is:
* apply the existing patchset with `hp -B glucas`
* edit gsetup.c
* `git commit gsetup.c` or `git commit --amend gsetup.c`
* (optional) make a backup of your existing patchset file, in case something goes wrong.
* regenerate the patchset file with `hp -e glucas`

Remember: a patchset file is very similar to what the `git format-patch --no-numbered --no-stat` command produces, so the only portions which may be edited by hand are the metadata in the headers.
Actually, a patchset containing e.g. 3 commits and generated with `hp -e` could also be obtained with:
```
git format-patch --no-numbered --no-stat --stdout HEAD~3 | sed "s/^Subject: \[PATCH\]/Subject:/"
```